lib/sync/storage/dataset-clients.js

```diff
@@ -52,12 +52,31 @@ function doListDatasetClients(filter, cb) {
  */
 function doRemoveDatasetClients(datasetClientsToRemove, cb) {
   var removeIds = _.pluck(datasetClientsToRemove, 'id');
-  debug('doRemoveDatasetClients removeIds = %j', removeIds);
-  async.map(removeIds, doRemoveDatasetClientWithRecords, function(err, deleteResult) {
+  var datasetIds = _.uniq(_.pluck(datasetClientsToRemove, 'datasetId'));
+  debug('doRemoveDatasetClients: removed datasetClients = %d, datasets = %d', removeIds.length, datasetIds.length);
+  async.series([
+    function deleteDatasetClientAndRefs(callback) {
+      async.map(removeIds, doRemoveDatasetClientWithRecords, function(err, deleteResult) {
+        if (err) {
+          debug('Failed to delete datasetClients due to error %j', err);
```

Comment:
I think this might be better to console.error() this so the developer knows something isn't working right and they should take an action from it.
That action is to probably raise a bug against fh-mbaas-api